File: backend/src/main/java/at/ac/tuwien/sepr/groupphase/backend/service/HallService.java
package at.ac.tuwien.sepr.groupphase.backend.service;

import at.ac.tuwien.sepr.groupphase.backend.endpoint.dto.HallCreateDto;
import at.ac.tuwien.sepr.groupphase.backend.entity.Hall;
import at.ac.tuwien.sepr.groupphase.backend.exception.ConflictException;

public interface HallService {
    /**
     * Creates a new hall.
     *
     * @param hallCreateDto data of the hall to create
     *
     * @throws ConflictException If the location of the DTO does not exist
     */
    Hall createHall(HallCreateDto hallCreateDto);
}

File: frontend/src/app/services/hall.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { Globals } from '../global/globals';
import { HallCreateDto } from '../dtos/hall-create-dto';

@Injectable({
  providedIn: 'root'
})
export class HallService {

  private baseUri: string = this.globals.backendUri + '/halls';

  constructor(private httpClient: HttpClient, private globals: Globals) {
  }

  /**
   * Create a new location and persist it in the backend
   * @param hallCreateDto
   */
  createHall(hallCreateDto: HallCreateDto): Observable<HallCreateDto> {
    return this.httpClient.post<HallCreateDto>(`${this.baseUri}`, hallCreateDto);
  }
}
